zebra-chain/src/transaction/unmined.rs

```diff
@@ -443,4 +455,28 @@ impl VerifiedUnminedTx {
 
         cost
     }
+
+    /// When deriving `Arbitrary` for `VerifiedUnminedTx` action fields can be generated
+    /// that are against consensus rules. This function prevent those issues by generating
+    /// new conventional and unpaid actions where rules are considered.
+    #[cfg(any(test, feature = "proptest-impl"))]
+    pub fn fix_arbitrary_generated_action_overflows(&mut self) -> Self {
+        let mut rng = rand::thread_rng();
+
+        // The number of actions can't be greater than 2^16.
+        let conventional_actions = rng.gen_range(0..2 ^ 16);
```

Comment:
In Rust, `^` means "bitwise exclusive or":
https://doc.rust-lang.org/std/ops/trait.BitXor.html

But the power of two operator is actually "bitwise shift left":
https://doc.rust-lang.org/std/ops/trait.Shl.html

Also the limit in the spec is `< 2^16`, rather than `<= 2^16`:
https://zips.z.cash/protocol/protocol.pdf#txnconsensus

```suggestion
        // The number of actions must be less than 2^16.
        let conventional_actions = rng.gen_range(0..(1 << 16));
```

Or you can use `u16::MAX`:

```suggestion
        // The number of actions must be less than 2^16.
        let conventional_actions = rng.gen_range(0..(u16::MAX as u32));
```